internal/config/config.go

```diff
@@ -281,24 +281,26 @@ func Load(cfg *Config) *viper.Viper {
 	}
 
 	if err = v.ReadInConfig(); err != nil {
-		log.Warn().Msgf("Not using config.yaml: %s", err.Error())
+		slog.Warn("Not using config.yaml", slog.Any("error", err))
 	}
 	if err = v.Unmarshal(cfg); err != nil {
-		log.Warn().Msgf("Mapping to configuration: %s", err.Error())
+		slog.Warn("Mapping to configuration", slog.Any("error", err))
 	}
 
 	return v
 }
 
 func reportError(err error) {
-	// TODO use logging framework
-	fmt.Fprint(os.Stderr, "Config validation error:\n")
 	for _, err := range err.(validator.ValidationErrors) {
-		fmt.Fprintf(
-			os.Stderr,
-			"    %s: rule: %v %v, got: '%v' (type: %v)\n",
-			err.Namespace(),
-			err.Tag(), err.Value(), err.Param(), err.Kind(),
+		slog.Error(
+			"Configuration validation error",
+			slog.String("namespace", err.Namespace()),
+			slog.Group("rule",
+				slog.String("tag", err.Tag()),
+				slog.String("value", fmt.Sprintf("%v", err.Value())),
```

Comment:
Is `slog.Any("value", err.Value())` equivalent to `%v` string formatting?

Reply:
Yes, it seems equivalent; I have just tried.

For the sake of curiosity, `any` is defined as the below:

```raw
// any is an alias for interface{} and is equivalent to interface{} in all ways.
type any = interface{}
```

Reply:
Fixed.

Reply:
The last change include this resolution!

Verified by using a `config.yaml` file comenting the `secret` line and the log was printed out with the new update.

Resolving this message.